projects/components/src/table/cells/data-renderers/string-array/string-array-table-cell-renderer.component.ts

```diff
@@ -43,20 +48,35 @@ import { TableCellAlignmentType } from '../../types/table-cell-alignment-type';
   alignment: TableCellAlignmentType.Left,
   parser: CoreTableCellParserType.NoOp,
 })
-export class StringArrayTableCellRendererComponent extends TableCellRendererBase<string[]> implements OnInit {
+export class StringArrayTableCellRendererComponent
+  extends TableCellRendererBase<string[], string[], StringArrayTableCellConfigOptions>
+  implements OnInit {
   public maxItemsInTooltip: number = 50;
-
+  protected viewMode: StringArrayTableCellDisplayMode = StringArrayTableCellDisplayMode.List;
   public constructor(
-    @Inject(TABLE_COLUMN_CONFIG) columnConfig: TableColumnConfig,
+    @Inject(TABLE_COLUMN_CONFIG) columnConfig: TableColumnConfig<StringArrayTableCellConfigOptions>,
     @Inject(TABLE_COLUMN_INDEX) index: number,
     @Inject(TABLE_DATA_PARSER) parser: TableCellParserBase<string[], string[], boolean>,
     @Inject(TABLE_CELL_DATA) cellData: string[],
     @Inject(TABLE_ROW_DATA) rowData: TableRow,
   ) {
     super(columnConfig, index, parser, cellData, rowData);
+
+    if (this.columnConfigOptions?.viewMode) {
+      this.viewMode = this.columnConfigOptions.viewMode;
+    }
```

Comment:
We should make `this.viewMode` readonly and
```ts
 this.viewMode = this.columnConfigOptions?.viewMode ?? StringArrayTableCellDisplayMode.List
```

Reply:
don't want to reassign and trigger cd if its not changing. made it readonly.

Reply:
I did not get you, it is not reassigning. It is the same as if condition.

Reply:
Since we're assigning in the constructor anyways

Reply:
if viewMode is empty, the variable will be reassigned with the same value ('list') again. This could trigger change detection since a reassignment happened. We can avoid it if we do the assignment only when value is supplied.